### ISSUE TITLE
A Question about Creating A New CCF Network

### ISSUE THREAD

**Issue by Silence-zyx** [microsoft/CCF#2800]
I refer to [Running a CCF Service](https://microsoft.github.io/CCF/ccf-1.0.6/operations/start_network.html) to create a new CCF network (version 1.0.6 )in a virtual mode in VM (Ubuntu 18.04).

I success to run it with commands (unimportant parameters ignored):

`./cchost.virtual  --enclave-file  ../liblogging.virtual.so ....`

The **/liblogging.virtual.so** is precompiled.

What confuses me is that if it exists a **universal virtual enclave lib** to run it.

In other words, do I have to specify an **App** when running a CCF network?

**Comment by eddyashton** [comment]
Hi @Silence-zyx. CCF is a framework, and `cchost` is a host launcher which can launch any CCF app instance, but that app instance must be specified as `--enclave-file`. This application defines the endpoints which are available on the running node. If you're looking for a more generic/universal application which allows you to modify the deployed endpoints at runtime (rather than through code upgrades and node cycling), you can use `js_generic` and write the application in JS. See [this discussion](https://github.com/microsoft/CCF/discussions/2704) for more context on the requirement for `--enclave-file`, and the difference between C++ and JS apps.

**Comment by achamayou** [comment]
@Silence-zyx we do welcome questions, and we tend to use [Discussions](https://github.com/microsoft/CCF/discussions) for them rather than Issues.
